src/main/cpp/src/hive_hash.cu

```diff
@@ -210,11 +210,72 @@ class hive_device_row_hasher {
       return this->hash_functor.template operator()<T>(col, row_index);
     }
 
+    struct StackElement{
+      cudf::column_device_view col;   // current column
+      int child_idx;                  // index of the child column to process next
+      int factor_exp;                 // factor exponent for the current column
+      //factor_exp = parent.factor_exp + parent.child_num - 1 - parent.child_idx
+
+      __device__ StackElement() = delete;
+      __device__ StackElement(cudf::column_device_view col, int factor_exp) : col(col), child_idx(0), factor_exp(factor_exp) {}
+    };
+
+    typedef StackElement* StackElementPtr;
     template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
     __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                             cudf::size_type row_index) const noexcept
     {
-      CUDF_UNREACHABLE("Nested type is not supported");
+      hive_hash_value_t ret = HIVE_INIT_HASH;
+      cudf::column_device_view curr_col = col.slice(row_index, 1);
+      // column_device_view default constructor is deleted, can not allocate column_device_view array directly
+      // use byte array to wrapper StackElement list
+      constexpr int len_of_8_StackElement = 8 * sizeof(StackElement);
+      uint8_t stack_wrapper[len_of_8_StackElement];
+      StackElementPtr stack = reinterpret_cast<StackElementPtr>(stack_wrapper);
```

Comment:
I am confused on creating a stack of StackElement ?
Does the below not work in C++ ?
```
StackElement statck[8];
```

Reply:
> I am confused on creating a stack of StackElement ? Does the below not work in C++ ?
> 
> ```
> StackElement statck[8];
> ```
Because if I define an array `StackElement stack[8];` without initialization, each element of the array will be constructed using the default constructor of `StackElement`. However, the default constructor for the member `cudf::column_device_view` in `StackElement` is deleted.

Reply:
OK, But you can restore the default constructor. Any reason it has to be deleted?
`uint8_t stack_wrapper[len_of_8_StackElement];` will also initialize 8 empty `StackElement`, the same as what `StackElement statck[8];` will do.